hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/impl/StreamDataChannelBase.java

```diff
@@ -93,6 +96,18 @@ public final boolean isOpen() {
     return getChannel().isOpen();
   }
 
+  protected void assertSpaceAvailability(int requested) throws StorageContainerException {
+    final HddsVolume volume = containerData.getVolume();
+    final SpaceUsageSource currentUsage = volume.getCurrentUsage();
+    final long spared = volume.getFreeSpaceToSpare(currentUsage.getCapacity());
+
+    if (currentUsage.getAvailable() - spared <= requested) {
+      throw new StorageContainerException("Failed to write " + requested + " bytes to container "
+          + containerData.getContainerID() + " due to volume " + volume.getStorageID() + " out of space "
+          + currentUsage + ", spared="  + spared, DISK_OUT_OF_SPACE);
+    }
+  }
```

Comment:
If you don't want to accept my [suggestion](https://github.com/apache/ozone/pull/8644#discussion_r2154304362) about moving this to `HddsVolume` with passing the container-specific part of the message as a parameter, then at least please move it to `ContainerUtils`.  It should be reused for non-streaming write in #8642.

Reply:
ok, moved to the ContainerUtils